### ISSUE TITLE
Weibull Adstock transformation results are different for training vs whole dataset

### ISSUE THREAD

**Issue by math-max** [facebookexperimental/Robyn#374]
## Describe issue
We have some questions about the Weibull Adstock function. My understanding of applying the Adstock transformation is that we're carrying over the effect of previous investment into the current time, therefore each value of the transformed variable gets adjusted using the previous time point value. In mathematical terms, 
`decay_{t,j} = x_{t,j} + f(x_{t-1,j}, param_{j})`
where `x_{t,k}` is the variable at time `t` and for media `j`, `f` is some function and `param` are some parameters for the function (theta, scale, shapes...) for media `j`. 

What we see is that when we pass to the Weibull Adstock function two datasets for the same media, one with only the training data and then a second one that includes more recent data (test set), the outputs are not matching. The behaviour we expected was that adding more data at the end of the time series, should only affect the last values, but what we actually see is that the transformed results are completely different even for the starting data-points. We think this is faulty behaviour as Adstock transformations should affect only the future and not the past, therefore adding extra data-points at the end of a time series should not affect the value for the transformed variable at the start. We've observed this behaviour only for the Weibull adstock, not for the Geometric. 

I would like to understand if this is an expected behaviour and if so, how could we apply this transformation for out-of-sample data. Thanks!

## Sample Code 
We've run a reproducible test of this behaviour using the simulated data provided with Robyn. Below the sample code, where we're applying the Adstock transformation to the Search spend, but in two different ways, first passing the whole dataset (208 observations), and then only providing the first 150 observations. 
```
shape <- 1.55
scale <- 0.04
m <- dt_simulated_weekly[, "search_S"]
ads_whole <- adstock_weibull(x = m$search_S, shape = shape, scale = scale)
ads_train <- adstock_weibull(x = m$search_S[1:150], shape = shape, scale = scale)
x_ads_whole <- ads_whole$x_decayed
x_ads_train <- ads_train$x_decayed
```
In both of these cases, only the first two values remain the same, but all other values of the transformed variable are different, see screenshot below. 
![image](https://user-images.githubusercontent.com/46674398/165072804-ae257dde-79a8-4b50-929f-1cae1e2eb6f3.png)

In the case of the Geometric adstock function, we're not seeing this type of results, the values are matching as expected. 
```
theta <- 0.4
m <- dt_simulated_weekly[, "search_S"]
ads_whole <- adstock_geometric(x = m$search_S, theta=theta)
ads_train <- adstock_geometric(x = m$search_S[1:150], theta=theta)
x_ads_whole <- ads_whole$x_decayed
x_ads_train <- ads_train$x_decayed
```
In this case, the results from both transformations are matching: 
![image](https://user-images.githubusercontent.com/46674398/165073858-5d76e316-f5b7-4d33-afe3-d44fe302ca69.png)

## Environment & Robyn version
R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 10 (buster)

Robyn version: Robyn_3.6.2

**Comment by kyletgoldberg** [comment]
Hi - thanks for the question and apologies for the delay in getting back to you. This is working as intended, since Weibull adstock (esp. scale param) is in fact dependent on the length of the dataset. I'll illustrate here with an example - 

![image](https://user-images.githubusercontent.com/84025306/165798182-2a1ac307-e82f-4eb8-9079-bdf58d8a0950.png)

Looking at this chart in the bottom left, where for example shape =9 and scale = 0.5. when m1 is 100 days, inflexion is at about day 40, or let's say around position 40% of X axis. When the user cuts the first 50 days for transformation, the relative position of inflexion point changes to day 20.

So the best way to work around your use case would be to do all of the adstocking transformation prior to doing the splitting of the data into train/test, so then there will be no issue with the window length anymore.
